magicctapipe/scripts/lst1_magic/lst1_magic_cloud_profile_estimator.py

```diff
@@ -0,0 +1,390 @@
+
+Usage:
+$ python lst1_magic_cloud_profile_estimator.py
+--input_file "/user/data/dl1_stereo_LST-1_MAGIC.Run03265.0040.h5"
+(--output_dir "/user/data/cloud_profiles/")
+(--config_file "/user/data/config.yaml")
+"""
+
+import argparse
+import logging
+import time
+from pathlib import Path
+
+import astropy.units as u
+import ctapipe
+import h5py
+import numpy as np
+import pandas as pd
+import tables
+import yaml
+from astropy.coordinates import AltAz, SkyCoord
+from ctapipe.coordinates import TelescopeFrame
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.io import read_table
+
+import magicctapipe
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def model(imp, h, alt):
+    """
+    Geometrical model needed to estimate the vertical tranmission profile of a cloud.
+    Calculates an oﬀset angle from the primary direction corresponding
+    to the emission height of the Cherenkov light need to construct the longitudinal
+    profile of the observed Cherenkov light
+    and to estimate the vertical tranmission profile of a cloud.
+
+    Parameters
+    ----------
+    imp : float
+        Stereo-reconstructed impact parameter of the event in units of meter.
+    h : float
+        Emission height of the Cherenkov light (above the observatory) in units of meter.
+    alt : float
+        Altitude angle of the observations in units of radian.
+
+    Returns
+    -------
+    float
+        Oﬀset angle (in united of degree) from the primary direction corresponding to
+        the emission height of the Cherenkov light.
+    """
+
+    d = h / np.sin(alt)
+    model = np.arctan((imp / d).to("")).to_value("deg")
+    correction_factor = 0.85 / np.sin(alt)
+    return model * correction_factor
+
+
+def make_profile(
+    telescopes_images,
+    dl1_params,
+    config,
+    assigned_tel_ids,
+    magic_only_analysis=False,
+):
+    """
+    Calculates the longitudinal distribution of the observed Cherenkov light.
+
+    Parameters
+    ----------
+    telescopes_images : dict
+        Dictionary containing telescope's details and image.
+    dl1_params : pandas.DataFrame
+        Pandas DataFrame containing DL1 and stereo-reconstruced parameters.
+    config : dict
+        Configuration dictionary containing various processing parameters, including those for
+        the estimation of the longitudinal distribution of the deteced Cherenkov light.
+    assigned_tel_ids : dict
+        Dictionary mapping telescope types to their respective telescope IDs.
+    magic_only_analysis : bool, optional
+        If `True`, the function reconstructs the stereo parameters using only MAGIC events (default is `False`).
+
+    Returns
+    -------
+    pandas.DataFrame
+        Updated dl1_params DataFrame including the longitudinal profile of the observed Cherenkov light.
```

Comment:
the original script produced separate output files with the aggregated longitudinal profile (summed up over all the events) from the whole file. 
The way how you did it now with saving a profile for each event is of course more flexible (in terms of applying any additional event selection, because for computing the transmission we either way need statistics of many events), but the output files will be pretty bulky and running over them could be troublesome. It can stay like this for the moment